onnxruntime/test/python/quantization/test_save_activations.py

```diff
@@ -0,0 +1,164 @@
+        self.preprocess_flag = True
+
+
+class TestSaveActivations(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_model_dir = tempfile.TemporaryDirectory(prefix="test_save_activations.")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._tmp_model_dir.cleanup()
+
+    def construct_test_model1(self, test_model_path):
+        #    (input)
+        #       |
+        #      Relu
+        #      /  \
+        #    Conv  \
+        #     |     \
+        #    Relu  Conv
+        #     |     |
+        #   Conv    |
+        #     \     /
+        #       Add
+        #        |
+        #       (X6)
+        input = helper.make_tensor_value_info("input", TensorProto.FLOAT, [1, 3, 1, 3])
+        x1_output = helper.make_tensor_value_info("X1", TensorProto.FLOAT, [1, 3, 1, 3])
+        x2_output = helper.make_tensor_value_info("X2", TensorProto.FLOAT, [1, 3, 1, 3])
+        x3_output = helper.make_tensor_value_info("X3", TensorProto.FLOAT, [1, 3, 1, 3])
+        x4_output = helper.make_tensor_value_info("X4", TensorProto.FLOAT, [1, 3, 1, 3])
+        x5_output = helper.make_tensor_value_info("X5", TensorProto.FLOAT, [1, 3, 1, 3])
+        x6_output = helper.make_tensor_value_info("X6", TensorProto.FLOAT, [1, 3, 1, 3])
+        w1 = generate_input_initializer([3, 3, 1, 1], np.float32, "W1")
+        b1 = generate_input_initializer([3], np.float32, "B1")
+        w3 = generate_input_initializer([3, 3, 1, 1], np.float32, "W3")
+        b3 = generate_input_initializer([3], np.float32, "B3")
+        w5 = generate_input_initializer([3, 3, 1, 1], np.float32, "W5")
+        b5 = generate_input_initializer([3], np.float32, "B5")
+        relu_node_1 = helper.make_node("Relu", ["input"], ["X1"], name="Relu1")
+        conv_node_1 = helper.make_node("Conv", ["X1", "W1", "B1"], ["X2"], name="Conv1")
+        relu_node_2 = helper.make_node("Relu", ["X2"], ["X3"], name="Relu2")
+        conv_node_2 = helper.make_node("Conv", ["X3", "W3", "B3"], ["X4"], name="Conv2")
+        conv_node_3 = helper.make_node("Conv", ["X1", "W5", "B5"], ["X5"], name="Conv3")
+        add_node = helper.make_node("Add", ["X4", "X5"], ["X6"], name="Add")
+
+        # we are keeping all tensors in the output anyway for verification purpose
+        graph = helper.make_graph(
+            [relu_node_1, conv_node_1, relu_node_2, conv_node_2, conv_node_3, add_node],
+            "test_graph_4",
+            [input],
+            [x1_output, x2_output, x3_output, x4_output, x5_output, x6_output],
+        )
+        graph.initializer.add().CopyFrom(w1)
+        graph.initializer.add().CopyFrom(b1)
+        graph.initializer.add().CopyFrom(w3)
+        graph.initializer.add().CopyFrom(b3)
+        graph.initializer.add().CopyFrom(w5)
+        graph.initializer.add().CopyFrom(b5)
+        model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 13)])
+        onnx.save(model, test_model_path)
+
+    def test_saved_act(self):
```

Comment:
I would include the expected behavior as part of the test method name. To make what a test is testing clear: `test_<method or function name>_<expected behavior>_[when_<condition>]`.

e.g. `test_method_x_raises_error_when_dims_is_not_a_sequence`

----

readability: Avoid abbreviations - they add cognitive load to readers.

> Do not use abbreviations that are ambiguous or unfamiliar to readers outside the project, and do not abbreviate by deleting letters within a word.

https://google.github.io/styleguide/pyguide.html#316-naming
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
fixed